src/Shortcodes/FileShortcodeProvider.php

```diff
@@ -139,11 +141,12 @@ public static function handle_shortcode($arguments, $content, $parser, $shortcod
     protected static function getCachedMarkup($cache, $cacheKey, $arguments): string
     {
         $item = $cache->get($cacheKey);
-        if ($item && !empty($item['filename'])) {
+        $assetStore = Injector::inst()->get(AssetStore::class);
+        if ($item && !empty($item['filename']) && $assetStore->exists($item['filename'], $item['hash'])) {
```

Comment:
```suggestion
        if ($item && !empty($item['filename'])) {
```
Move this below - we shouldn't hit the filesystem if we're not even allowing the grant.

Reply:
Updated